File: examples/blobs/doc-blob-list-paginate.swift
import JsBaoClient

// Page through a document's blobs by following the opaque `cursor`. The cursor
// is only present when more results exist.
func paginateDocumentBlobs(
  client: JsBaoClient,
  documentId: String
) async throws {
  let blobs = client.documents.blobs(documentId: documentId)
  // #region example
  let page1 = try await blobs.list(limit: 50)
  for b in page1.items {
    print(b.blobId, b.filename, b.contentType, b.numBytes, b.sha256, b.createdAt)
  }

  // `cursor` is an opaque token; only present when more results remain.
  if let cursor = page1.cursor {
    let page2 = try await blobs.list(cursor: cursor)
    _ = page2.items
  }
  // #endregion example
}

File: examples/blobs/doc-blob-list-paginate.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Page through a document's blobs by following the opaque `cursor`. The cursor
// is only present when more results exist.
export async function paginateDocumentBlobs(
  client: JsBaoClient,
  documentId: string,
) {
  const blobs = client.document(documentId).blobs();
  // #region example
  const page1 = await blobs.list({ limit: 50 });
  for (const b of page1.items) {
    console.log(b.blobId, b.filename, b.contentType, b.numBytes, b.sha256, b.createdAt);
  }

  // `cursor` is an opaque token; only present when more results remain.
  if (page1.cursor) {
    const page2 = await blobs.list({ cursor: page1.cursor });
    return page2.items;
  }
  // #endregion example
  return page1.items;
}
